### PR TITLE
T#1523/next prev skill buttons

### PR DESCRIPTION
**Pull request by dwalizer**
Per #1523 - add Next / Prev buttons to the Skill overview page

### CODE REVIEW

File: service/src/main/java/skills/skillLoading/SkillsLoader.groovy [hunk 1]
```diff
@@ -406,10 +410,25 @@ class SkillsLoader {
...
-    SkillSummary loadSkillSummary(String projectId, String userId, String crossProjectId, String skillId) {
+    SkillSummary loadSkillSummary(String projectId, String userId, String crossProjectId, String skillId, String subjectId) {
         ProjDef projDef = getProjDef(userId, crossProjectId ?: projectId)
         SkillDefWithExtra skillDef = getSkillDefWithExtra(userId, crossProjectId ?: projectId, skillId, [SkillDef.ContainerType.Skill, SkillDef.ContainerType.SkillsGroup])
 
+        String nextSkillId;
+        String prevSkillId;
+
+        if(subjectId) {
+            SkillSubjectSummary subject = loadSubject(projectId, userId, subjectId, publicProps.getInt(PublicProps.UiProp.maxSkillVersion), true)
```
Comment (sudo-may) [inline service/src/main/java/skills/skillLoading/SkillsLoader.groovy:421]: ``loadSubject`` method is very expensive; suggesting to execute 2 sql statements to locate prev and next skillIds (1 for previous and 1 for next skillId)

File: e2e-tests/cypress/e2e/client-display/client-display-skill-paging_spec.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+describe('Client Display Skills Navigation', () => {
+
+    it('navigate between skills', () => {
```
Comment (sudo-may) [inline e2e-tests/cypress/e2e/client-display/client-display-skill-paging_spec.js:18]: this test looks great! please add another tests that starts on a Skill page and does similar type of prev/next navigation

**Review by sudo-may — CHANGES_REQUESTED** [review]

File: service/src/main/java/skills/skillLoading/SkillsLoader.groovy (hunk 1 above)
Comment (dwalizer) [reply service/src/main/java/skills/skillLoading/SkillsLoader.groovy]: Changed to a new method that runs 2 SQL statements

File: e2e-tests/cypress/e2e/client-display/client-display-skill-paging_spec.js (hunk 2 above)
Comment (dwalizer) [reply e2e-tests/cypress/e2e/client-display/client-display-skill-paging_spec.js:18]: Added more tests to the spec

**Pull request merged**